### ISSUE TITLE
Crashes on promise rejection response

### ISSUE THREAD

**Issue by CarterLi** [Romakita/ts-express-decorators#96]
## Informations

Type | Version
---|---
Bug | 2.x ( tested 2.0.13)

****

## Description

I tries to fetch another rest API using [axios](https://github.com/mzabriskie/axios), which returns a promise.

```ts
@Get('/test')
public async test() {
  try {
    const { data } = await axios.get(`http://0.0.0.0`);
    return data;
  } catch (e) {
    return e;
  }
}
```

The promise is rejected with `Error: connect ECONNREFUSED 0.0.0.0:80`. I expect something like 500 but it turns out that it server crashes internally in `ts-express-decorators` and clients cannot get any response.

```
TypeError: Converting circular structure to JSON
    at JSON.stringify (<anonymous>)
    at info (/Users/Carter/platform-management-server/node_modules/ts-express-decorators/src/mvc/class/HandlerBuilder.ts:138:39)
    at Object.locals.next (/Users/Carter/platform-management-server/node_modules/ts-express-decorators/src/mvc/class/HandlerBuilder.ts:155:58)
    at HandlerBuilder.<anonymous> (/Users/Carter/platform-management-server/node_modules/ts-express-decorators/src/mvc/class/HandlerBuilder.ts:179:20)
    at step (/Users/Carter/platform-management-server/node_modules/ts-express-decorators/lib/mvc/class/HandlerBuilder.js:40:23)
    at Object.throw (/Users/Carter/platform-management-server/node_modules/ts-express-decorators/lib/mvc/class/HandlerBuilder.js:21:53)
    at rejected (/Users/Carter/platform-management-server/node_modules/ts-express-decorators/lib/mvc/class/HandlerBuilder.js:13:65)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

`express-decorators/src/mvc/class/HandlerBuilder.ts:138:39` points to 

```ts
const info = (o = {}) => JSON.stringify({
    type: this.handlerMetadata.type,
    target: (target ? nameOf(target) : target.name) || "anonymous",
    methodName,
    injectable,
    data: dataStored,
    ...o
});
```

Note `GlobalErrorHandlerMiddleware` is injected in `ServerLoader` but it doesn't work.

## Todos

- [x] Example (log, usecase, etc...)
- [x] Ts.ED version (bug/question)

**Comment by Romakita** [comment]
Hi @CarterLi,

I'll fix that ASAP ;)

In addition, (and after the fix will be released) the try catch isn't necessary because Ts.ED will catch the error thrown by axio. Return an error isn't the good way because, Ts.ED will understand that is a good response and not an internal server error.
 You can do that:
```typescript
@Get('/test')
public async test() {
    const { data } = await axios.get(`http://0.0.0.0`); // Ts.ED handle this error and reject it (normally)
}
```
OR
```typescript
const {InternalServerError} from "ts-httpexceptions";
@Get('/test')
public async test() {
  try {
    const { data } = await axios.get(`http://0.0.0.0`);
    return data;
  } catch (e) {
    throw new InternalServerError("Gateway is down"); // or custom Error
  }
}
```

See you,
Romain

**Comment by Romakita** [comment]
Fixed with v2.0.14

**Comment by CarterLi** [comment]
> In addition, (and after the fix will be released) the try catch isn't necessary because Ts.ED will catch the error thrown by axio.

Yes. I incorrectly copy&pasted my dirty fix. It shouldn't be there.

Thanks for your quick fix 👍
